### PR TITLE
Platform 1946 signature single account flag

### PR DESCRIPTION
**Pull request by erpankajpatel**
- signature changes and sing account flag

### CODE REVIEW

File: android/src/main/java/com/phylloconnect/PhylloConnectModule.kt [hunk 1]
```diff
@@ -73,7 +68,18 @@ class PhylloConnectModule(reactContext: ReactApplicationContext) : ReactContextB
...
+            }
+
+            var map = hashMapOf<String, Any?>(
+            "clientDisplayName" to name,
+            "token" to token,
+            "workPlatformId" to platformId,
+            "userId" to userId,
+            "environment" to  getPhylloEnvironment(environment),
+            "callback" to callback,
+            "singleAccount" to singleAccount
```
Comment (akshaybhalotia) [inline android/src/main/java/com/phylloconnect/PhylloConnectModule.kt:80]: Same comment - shouldn't be exposed

File: example/src/App.js [hunk 2]
```diff
@@ -68,6 +68,7 @@ export default function ExampleApp() {
     const clientDisplayName = 'Example'
     const externalId = generateRandomString(20)
     const environment = clientConfig.env
+    const singleAccount = true
```
Comment (akshaybhalotia) [inline example/src/App.js:71]: Shouldn't be exposed

File: example/src/App.js [hunk 3]
```diff
@@ -92,6 +93,7 @@ export default function ExampleApp() {
         userId: id,
         workPlatformId,
         environment,
+        singleAccount
```
Comment (akshaybhalotia) [inline example/src/App.js:96]: Shouldn't be exposed

File: ios/PhylloConnectModule.m [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 @interface RCT_EXTERN_MODULE(PhylloConnectModule, NSObject)
 
-RCT_EXTERN_METHOD(initialize:clientDisplayName:token:userId:environment:workPlatformId)
+RCT_EXTERN_METHOD(initialize:(NSString *)clientDisplayName:(NSString *)token:(NSString *)userId:(NSString *)environment:(NSString *)workPlatformId:(BOOL *)singleAccount)
```
Comment (akshaybhalotia) [inline ios/PhylloConnectModule.m:5]: Shouldn't be exposed

File: ios/PhylloConnectModule.swift [hunk 5]
```diff
@@ -34,19 +34,20 @@ public class PhylloConnectModule: RCTEventEmitter {
     }
     
     
-    @objc(initialize:::::)
-    func initialize(clientDisplayName:String,token:String,userId:String ,environment:String, workPlatformId:String) {
+    @objc(initialize::::::)
+    func initialize(clientDisplayName:String,token:String,userId:String ,environment:String, workPlatformId:String,singleAccount:Bool = false) {
```
Comment (akshaybhalotia) [inline ios/PhylloConnectModule.swift:38]: Shouldn't be exposed

File: ios/PhylloConnectModule.swift [hunk 6]
```diff
@@ -34,19 +34,20 @@ public class PhylloConnectModule: RCTEventEmitter {
...
-                                        )
-            PhylloConnect.shared.initialize(config: phylloConfig)
+        var phylloConfig = [String:Any]()
+        phylloConfig["environment"] = self.getEnvironment(env: environment)
+        phylloConfig["clientDisplayName"] = clientDisplayName
+        phylloConfig["token"] = token
+        phylloConfig["userId"] = userId
+        phylloConfig["delegate"] = self
+        phylloConfig["workPlatformId"] = workPlatformId
+        phylloConfig["singleAccount"] = singleAccount
```
Comment (akshaybhalotia) [inline ios/PhylloConnectModule.swift:48]: Shouldn't be exposed

File: src/PhylloConnect.ts [hunk 7]
```diff
@@ -14,6 +14,7 @@ interface IPhylloInitialize {
   userId: string
   environment: PhylloEnvironment
   workPlatformId?: string
+  singleAccount?: Boolean
```
Comment (akshaybhalotia) [inline src/PhylloConnect.ts:17]: Shouldn't be exposed

File: src/PhylloConnect.ts [hunk 8]
```diff
@@ -143,6 +144,7 @@ const PhylloConnectSDK = {
     userId,
     environment,
     workPlatformId = '',
+    singleAccount,
```
Comment (akshaybhalotia) [inline src/PhylloConnect.ts:147]: Shouldn't be exposed

File: src/PhylloConnect.ts [hunk 9]
```diff
@@ -157,7 +159,8 @@ const PhylloConnectSDK = {
       token,
       userId,
       environment,
-      workPlatformId
+      workPlatformId,
+      singleAccount
```
Comment (akshaybhalotia) [inline src/PhylloConnect.ts:163]: Shouldn't be exposed

**Review by akshaybhalotia — CHANGES_REQUESTED** [review]

**Review by akshaybhalotia — APPROVED** [review]

**Review by chiragphyllo — APPROVED** [review]

**Pull request merged**